vividus-plugin-rest-api/src/main/java/org/vividus/http/ExtendedHttpLoggingInterceptor.java

```diff
@@ -0,0 +1,142 @@
+        this.logger = LoggerFactory.getLogger(loggerType);
+        this.extendedLogging = extendedLogging;
+    }
+
+    @Override
+    public void process(HttpRequest request, EntityDetails entityDetails, HttpContext context) throws IOException
+    {
+        String argBrackets = " {}";
+        LoggingEventBuilder loggingEventBuilder = logger.atInfo();
+        StringBuilder loggerFormat = new StringBuilder("Request:");
+
+        ProtocolVersion protocolVersion = request.getVersion();
+        if (protocolVersion != null)
+        {
+            loggingEventBuilder = loggingEventBuilder.addArgument(protocolVersion);
+            loggerFormat.append(argBrackets);
+        }
+        loggingEventBuilder = loggingEventBuilder.addArgument(request);
+        loggerFormat.append(argBrackets);
+
+        if (extendedLogging)
+        {
+            Header[] headers = request.getHeaders();
+            String headersAsString = Stream.of(headers).map(Object::toString).collect(Collectors.joining(NEW_LINE));
+            loggingEventBuilder = loggingEventBuilder.addArgument(headersAsString);
+            loggerFormat.append(HEADERS_FORMAT);
+
+            if (request instanceof HttpEntityContainer httpEntityContainer)
+            {
+                HttpEntity entity = httpEntityContainer.getEntity();
+                if (entity instanceof StringEntity stringEntity)
+                {
+                    String textBody = new String(stringEntity.getContent().readAllBytes(), StandardCharsets.UTF_8);
+                    loggingEventBuilder = loggingEventBuilder.addArgument(textBody);
+                    loggerFormat.append(BODY_FORMAT);
+                }
+            }
+        }
+        loggingEventBuilder.log(loggerFormat.toString());
+    }
+
+    @Override
+    public void handle(HttpResponse httpResponse)
+    {
+        LoggingEventBuilder loggingEventBuilder = logger.atInfo().addArgument(httpResponse.getStatusCode())
+                .addArgument(httpResponse.getFrom());
+        StringBuilder loggerFormat = new StringBuilder("Response: status code {}, {}");
+
+        if (extendedLogging)
+        {
+            Header[] headers = httpResponse.getResponseHeaders();
+            String headersAsString = Stream.of(headers).map(Object::toString).collect(Collectors.joining(NEW_LINE));
+            loggingEventBuilder = loggingEventBuilder.addArgument(headersAsString);
+            loggerFormat.append(HEADERS_FORMAT);
+
+            String mimeType = getMimeType(headers).orElseGet(ContentType.DEFAULT_TEXT::getMimeType);
+            if (LOGGED_CONTENT_TYPES.contains(mimeType) && httpResponse.getResponseBody() != null)
+            {
+                loggingEventBuilder = loggingEventBuilder.addArgument(httpResponse.getResponseBodyAsString());
+                loggerFormat.append(BODY_FORMAT);
+            }
+        }
+        loggingEventBuilder.log(loggerFormat.toString());
+    }
+
+    private Optional<String> getMimeType(Header... headers)
```

Comment:
the logic is a partial copy of `PublishingAttachmentInterceptor`, please refactor to avoid duplications